env/flink/env_flink.cc

```diff
@@ -1,10 +1,854 @@
+    return std::string("Exception when GetFileModificationTime, file name: ")
+        .append(file_name);
+  });
+  if (!status.ok()) {
+    return status;
+  }
+
+  *time = fileModificationTime;
+  return IOStatus::OK();
+}
+
+IOStatus FlinkFileSystem::IsDirectory(const std::string& path,
+                                      const IOOptions& options, bool* is_dir,
+                                      IODebugContext* dbg) {
+  JNIEnv* jniEnv = getJNIEnv();
+  jobject fileStatus;
+  IOStatus status = GetFileStatus(path, options, dbg, &fileStatus);
+  if (!status.ok()) {
+    return status;
+  }
+
+  JavaClassCache::JavaMethodContext isDirMethod =
+      class_cache_->GetJMethod(JavaClassCache::JM_FLINK_FILE_STATUS_IS_DIR);
+  jboolean isDir =
+      jniEnv->CallBooleanMethod(fileStatus, isDirMethod.javaMethod);
+  jniEnv->DeleteLocalRef(fileStatus);
+
+  status = CurrentStatus([path]() {
+    return std::string("Exception when IsDirectory, file name: ").append(path);
+  });
+  if (!status.ok()) {
+    return status;
+  }
+
+  *is_dir = isDir;
+  return IOStatus::OK();
+}
+
+IOStatus FlinkFileSystem::RenameFile(const std::string& src,
+                                     const std::string& target,
+                                     const IOOptions& options,
+                                     IODebugContext* dbg) {
+  IOStatus status = FileExists(src, options, dbg);
+  if (!status.ok()) {
+    return status.IsNotFound()
+               ? IOStatus::PathNotFound(
+                     std::string(
+                         "Could not find src path when RenameFile, path: ")
+                         .append(ConstructPath(src)))
+               : status;
+  }
+
+  std::string srcFilePath = ConstructPath(src);
+  // Construct src Path Instance
+  jobject srcPathInstance;
+  status = class_cache_->ConstructPathInstance(srcFilePath, &srcPathInstance);
+  if (!status.ok()) {
+    return status;
+  }
+
+  std::string targetFilePath = ConstructPath(target);
+  // Construct target Path Instance
+  jobject targetPathInstance;
+  status =
+      class_cache_->ConstructPathInstance(targetFilePath, &targetPathInstance);
+  if (!status.ok()) {
+    return status;
+  }
+
+  JNIEnv* jniEnv = getJNIEnv();
+  JavaClassCache::JavaMethodContext renameMethod = class_cache_->GetJMethod(
+      JavaClassCache::JM_FLINK_FILE_SYSTEM_RENAME_FILE);
+  jboolean renamed =
+      jniEnv->CallBooleanMethod(file_system_instance_, renameMethod.javaMethod,
+                                srcPathInstance, targetPathInstance);
+  jniEnv->DeleteLocalRef(srcPathInstance);
+  jniEnv->DeleteLocalRef(targetPathInstance);
+
+  status = CurrentStatus([srcFilePath, targetFilePath]() {
+    return std::string("Exception when RenameFile, src: ")
+        .append(srcFilePath)
+        .append(", target: ")
+        .append(targetFilePath);
+  });
+  if (!status.ok()) {
+    return status;
+  }
+
+  return renamed ? IOStatus::OK() : IOStatus::IOError(status.ToString());
```

Comment:
The "status.ToString" always return "OK". Maybe we also need to construct the error message manually here.